go-controller/pkg/node/gateway_init.go

```diff
@@ -189,7 +189,32 @@ func getGatewayNextHops() ([]net.IP, string, error) {
 	if needIPv4NextHop || needIPv6NextHop || gatewayIntf == "" {
 		defaultGatewayIntf, defaultGatewayNextHops, err := getDefaultGatewayInterfaceDetails(gatewayIntf, config.IPv4Mode, config.IPv6Mode)
 		if err != nil {
-			return nil, "", err
+			if !(strings.Contains(err.Error(), "mistmaching provided gw interface") && config.Gateway.Mode == config.GatewayModeLocal && config.Gateway.AllowNoUplink) {
```

Comment:
I would prefer if we make this a new error type and then check if it is that error rather than checking strings